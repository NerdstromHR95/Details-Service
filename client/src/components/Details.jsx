import React from 'react';
import axios from 'axios';
import SizeSelect from './SizeSelect.jsx'
import ColorSelect from './ColorSelect.jsx'


class Details extends React.Component {
  constructor() {
    super();
    this.state = {
      product: {},
      colors: [],
      sizes: [{}, {}, {}],
    };
    this.getitem = this.getItem.bind(this);
    this.getSizes = this.getSizes.bind(this);
    this.getColors = this.getColors.bind(this);
  }

  getItem() {
    axios.get('http://localhost:3000/item')
      .then(res => {
        this.setState({
          product: res.data[0]
        });
      })
      .catch(function(err) {
        console.log('An error occurred: ', err);
      });
  }

  getSizes() {
    axios.get('http://localhost:3000/sizes')
      .then(res => {
        this.setState({
          sizes: res.data
        });
      })
      .catch(err => {
        console.log('An error occurred: ', err);
      });
  }

  getColors() {
    axios.get('http://localhost:3000/colors')
      .then(res => {
        this.setState({
          colors: res.data
        });
      })
      .catch(err => {
        console.log('An error occurred: ', err);
      });
  }

  componentDidMount() {
    this.getItem();
  }

  // componentDidUpdate() {
  //   this.getItem();
  //   this.getColors();
  // }

  render() {
    console.log(this.state.sizes)
    return (
      <div className='main'>  
        <div className='title'>
          {this.state.product.title} 
        </div>
        <div className='brand'>
          {this.state.product.brand}
        </div>
        <div className='price'>
          <strong>{this.state.product.price}</strong><span className='tab'>Free Shipping</span>
        </div>
        <div className='description'>
          {this.state.product.description}
        </div>
        <div className='sizes'>
          <div>
            <strong>Fit</strong>  <span className='small'>True to size.</span>
          </div>
          <SizeSelect />
        </div>
        <div className='menuBuffer'></div>
        <div className='colors'>
          <ColorSelect />
        </div>
      </div>
    );
  }
}

export default Details;

