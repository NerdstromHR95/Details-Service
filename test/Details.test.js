import React from 'react';
import Details from '../client/src/components/Details.jsx';
import renderer from 'react-test-renderer';

test('Should toggle size menu', () => {
	const component = renderer.create(<Details />);
	let tree = component.toJSON();
	
	expect(tree).toMatchSnapshot();

	tree.children[0].children[2].children[6].children[1].props.onClick();

	tree = component.toJSON();

	expect(tree).toMatchSnapshot();
})