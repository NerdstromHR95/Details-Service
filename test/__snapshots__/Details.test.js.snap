// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should toggle size menu 1`] = `
<div>
  <div
    className={null}
  >
    <div
      className="topPlaceholder"
    >
      Place Holder
    </div>
    <div
      className="rightPlaceholder"
    >
      Place Holder
    </div>
    <div
      className="main"
    >
      <div
        className="starRating"
        onClick={[Function]}
      >
        <div
          className="starRating"
        >
          <i
            className="fas fa-star-half"
          />
        </div>
      </div>
      <div
        className="title"
        onClick={[Function]}
      />
      <div
        className="brand"
        onClick={[Function]}
      />
      <div
        className="price"
        onClick={[Function]}
      >
        <strong />
        <span
          className="tab"
        >
          Free Shipping
        </span>
      </div>
      <div
        className="description"
        onClick={[Function]}
      />
      <div
        className="menuBuffer"
        onClick={[Function]}
      />
      <div
        className="sizes"
      >
        <div>
          <strong
            className="darkGrey"
          >
            Fit
          </strong>
            
          <span
            className="small"
          >
            True to size.
          </span>
        </div>
        <div
          className="selectedSize"
          onClick={[Function]}
        >
          <span
            className="smallIndent menuDefault"
          >
            Size
          </span>
          <i
            className="fas fa-angle-down arrow"
          />
        </div>
      </div>
      <div
        className="menuBuffer"
        onClick={[Function]}
      >
         Size Guide
      </div>
      <div
        className="colors"
      >
        <div>
          <div
            className="selectedColor"
            onClick={[Function]}
          >
            <span
              className="smallIndent menuDefault"
            >
              Color
            </span>
            <i
              className="fas fa-angle-down arrow"
            />
          </div>
        </div>
        <div
          className="colorSwatches"
        >
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Black"
              id="Black"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="White"
              id="White"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Blue"
              id="Blue"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Red"
              id="Red"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Green"
              id="Green"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
        </div>
      </div>
      <div
        className="formField"
      >
        <form>
          <input
            className="inputField"
            onChange={[Function]}
            placeHolder="1"
            type="text"
          />
          <div
            className="addButton"
            onClick={[Function]}
          >
            Add to Bag
          </div>
          <div
            className="wishButton"
            onClick={[Function]}
          >
            Add to Wishlist
          </div>
        </form>
      </div>
      <div
        className="bottomDiv"
      >
        Available for pickup in multiple sizes near you.
        <br />
        <div
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          See details
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should toggle size menu 2`] = `
<div>
  <div
    className={null}
  >
    <div
      className="topPlaceholder"
    >
      Place Holder
    </div>
    <div
      className="rightPlaceholder"
    >
      Place Holder
    </div>
    <div
      className="main"
    >
      <div
        className="starRating"
        onClick={[Function]}
      >
        <div
          className="starRating"
        >
          <i
            className="fas fa-star-half"
          />
        </div>
      </div>
      <div
        className="title"
        onClick={[Function]}
      />
      <div
        className="brand"
        onClick={[Function]}
      />
      <div
        className="price"
        onClick={[Function]}
      >
        <strong />
        <span
          className="tab"
        >
          Free Shipping
        </span>
      </div>
      <div
        className="description"
        onClick={[Function]}
      />
      <div
        className="menuBuffer"
        onClick={[Function]}
      />
      <div
        className="sizes"
      >
        <div>
          <strong
            className="darkGrey"
          >
            Fit
          </strong>
            
          <span
            className="small"
          >
            True to size.
          </span>
        </div>
        <div
          className="menu"
        >
          <div
            className="sizeOption"
            onClick={[Function]}
          >
            <strong
              className="smallIndent"
            >
              Please Choose A Size
            </strong>
          </div>
          <div
            className="sizeOption"
            id="small"
            onClick={[Function]}
          >
            small
          </div>
          <div
            className="sizeOption"
            id="medium"
            onClick={[Function]}
          >
            medium
          </div>
          <div
            className="sizeOption"
            id="large"
            onClick={[Function]}
          >
            large
          </div>
        </div>
      </div>
      <div
        className="menuBuffer"
        onClick={[Function]}
      >
         Size Guide
      </div>
      <div
        className="colors"
      >
        <div>
          <div
            className="selectedColor"
            onClick={[Function]}
          >
            <span
              className="smallIndent menuDefault"
            >
              Color
            </span>
            <i
              className="fas fa-angle-down arrow"
            />
          </div>
        </div>
        <div
          className="colorSwatches"
        >
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Black"
              id="Black"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="White"
              id="White"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Blue"
              id="Blue"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Red"
              id="Red"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
          <svg
            className="circle"
            height="35"
            width="35"
          >
            <circle
              className="backCircle"
              cx="20"
              cy="20"
              fill="black"
              r="15"
              stroke="#e3e4e5"
              strokeWidth="5"
            />
            <circle
              className="frontCircle"
              cx="20"
              cy="20"
              fill="Green"
              id="Green"
              onClick={[Function]}
              r="15"
              stroke="white"
              strokeWidth="4"
            />
          </svg>
        </div>
      </div>
      <div
        className="formField"
      >
        <form>
          <input
            className="inputField"
            onChange={[Function]}
            placeHolder="1"
            type="text"
          />
          <div
            className="addButton"
            onClick={[Function]}
          >
            Add to Bag
          </div>
          <div
            className="wishButton"
            onClick={[Function]}
          >
            Add to Wishlist
          </div>
        </form>
      </div>
      <div
        className="bottomDiv"
      >
        Available for pickup in multiple sizes near you.
        <br />
        <div
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          See details
        </div>
      </div>
    </div>
  </div>
</div>
`;
